src/librustc_mir/util/elaborate_drops.rs

```diff
@@ -675,7 +675,7 @@ where
         let tcx = self.tcx();
 
         if let Some(size) = opt_size {
-            let size: u32 = size.try_into().unwrap_or_else(|_| {
+            let size: u64 = size.try_into().unwrap_or_else(|_| {
                 bug!("move out check isn't implemented for array sizes bigger than u32::MAX");
```

Comment:
```suggestion
                bug!("move out check isn't implemented for array sizes bigger than u64::MAX");
```
That's what `check_if_subslice_element_is_moved` does, right?

Reply:
This entire check can be removed, `opt_size` is `Option<u64>` so the `size` from `if let Some(size)` is already `u64`.